### PR TITLE
1344 Typescript conversion - Card, Heading, IconCard

### PR DESCRIPTION
**Pull request by emil-botha**
- Convert the Card, Heading and IconCard components from Javascript to TypeScript
- Use functional instead of class components
- Implement initial unit tests

### CODE REVIEW

File: packages/bappo-components/src/components/IconCard/IconCard.tsx [hunk 1]
```diff
@@ -32,39 +17,38 @@ const IconCard = ({
...
 
-IconCard.defaultProps = {};
-
 export default IconCard;
 
 const Container = styled(TouchableView)`
   align-items: center;
-  width: ${(props) => `${props.$size}px` || '40px'};
-  margin: ${(props) => `${props.$size / 10}px` || '8px'};
+  width: ${({ $size }) => `${$size}px` || '40px'};
```
Comment (xuan9230) [inline packages/bappo-components/src/components/IconCard/IconCard.tsx]:
```suggestion
  width: ${({ $size }) => $size ? `${$size}px` : '40px'};
```

File: packages/bappo-components/src/components/IconCard/IconCard.tsx [hunk 2]
```diff
@@ -32,39 +17,38 @@ const IconCard = ({
...
+  margin: ${({ $size }) => `${$size / 10}px` || '8px'};
```
Comment (xuan9230) [inline packages/bappo-components/src/components/IconCard/IconCard.tsx]:
```suggestion
  margin: ${({ $size }) => $size ? `${$size / 10}px` : '8px'};
```

File: packages/bappo-components/src/components/IconCard/IconCard.tsx [hunk 3]
```diff
@@ -73,15 +57,15 @@ const StyledView = styled(View)`
...
   align-items: center;
   justify-content: center;
   margin: 8px;
 `;
 
-const StyledIcon = styled(Icon)`
-  width: ${(props) => `${(props.$size * 2) / 5}px` || '16px'};
-  font-size: ${(props) => `${props.$size / 3}px` || '14px'};
+const StyledIcon = styled(Icon)<{ $size: any }>`
+  width: ${({ $size }) => `${($size * 2) / 5}px` || '16px'};
```
Comment (xuan9230) [inline packages/bappo-components/src/components/IconCard/IconCard.tsx]: Similar issue as above

**Review by xuan9230 — CHANGES_REQUESTED** [review]
Hi, I've suggested changes and commented. Other than those it looks perfect

File: packages/bappo-components/src/components/IconCard/IconCard.tsx (hunk 3 above)
Comment (emil-botha) [reply packages/bappo-components/src/components/IconCard/IconCard.tsx]: Thanks for the suggestions! I've committed a change for this.

**Pull request merged**